### PR TITLE
fixed creator tag entering inside card on smaller devices and also fi…

### LINKED ISSUE #364: Horizontal overflow and horizontal Scroll on smaller devices when logged in.
Describe the bug
The page does not fit to screen on smaller devices when logged in.

**To Reproduce**

1. Go to zuhub homepage
2. login
3. use the developer tools or inspect on browser. (Ctrl + shift + I)

**Expected behavior**
page is supposed to  fit to device width without any overflow

**Screenshots**
Screenshot (107)

### LINKED ISSUE #365: Creator text on project cards disappearing into card on smaller screen devices
**Describe the bug**
The creator text goes into the card on smaller devices

**To Reproduce**

1. Go to 'http://localhost:3000/profile'
2. Use developer tools to adjust screen width
3. Scroll down to created projects'
4. notice the text entering inside the card

**Expected behavior**
the text should wrap to a new line when the width is exhausted

**Screenshots**
Screenshot (108)

### PR DESCRIPTION
**Pull request by Morai-C**
…xed horizontal scroll and overflow

## Summary

Description of PR here...
Closes #364, Closes #365

## Changes

adjusted styling for card to fit creator tag on small screen devices
included breakpoints in styling to fix problem of horizontal scroll and overflow on smaller devices

## Screenshots

Screenshot (112)
Screenshot (113)

 @tuxology please this is ready for review please check and also merge

### CODE REVIEW

File: zubhub_frontend/zubhub/package.json [hunk 1]
```diff
@@ -23,10 +23,10 @@
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-i18next": "^11.8.5",
-    "react-quill": "^1.3.5",
+    "react-quill": "^0.0.2",
```
Comment (tuxology) [inline zubhub_frontend/zubhub/package.json]: Any reason, why we downgraded this package?

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js [hunk 2]
```diff
@@ -1,6 +1,12 @@
+/* eslint-disable no-dupe-keys */
 const styles = theme => ({
   root: {
-    maxWidth: 345,
+    [theme.breakpoints.down('md')]: {
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: What does this do? Maybe I don't know. Just trying to learn. @NdibeRaymond probably can give us some insight

**Review by tuxology — COMMENTED** [review]
Need someone else to look at it also

File: zubhub_frontend/zubhub/package.json (hunk 1 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/package.json]: in order to access auto update on save i used npm to start the server on my terminal it was downgraded when i ran npm install command because it wasn't compatible with my version of npm. but anyother person who uses npm start on the frontend it's going to install the packages compatible with their version.

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 2 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: @tuxology  okay well Material UI also has media query breakpoints just like bootstrap so --> .down('md') <-- means on screens from medium sizes to extra small it should apply that style to them. basically the margins were causing the horizontal overflow on medium devices downwards so i removed them on those devices with the them.breakpoints styling..

File: zubhub_frontend/zubhub/package.json (hunk 1 above)
Comment (NdibeRaymond) [reply zubhub_frontend/zubhub/package.json]: @Morai-C , perhaps you should upgrade your npm version?

File: zubhub_frontend/zubhub/package.json [hunk 3]
```diff
@@ -23,10 +23,10 @@
...
     "react-redux": "^7.2.2",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3",
+    "react-scripts": "^5.0.0",
```
Comment (NdibeRaymond) [inline zubhub_frontend/zubhub/package.json]: it also looks like you upgraded the react script version here. This is not related to the changes you are working on. Like you said, if this was imposed on you by your npm installation, perhaps you should upgrade (or downgrade) your npm installation?

**Review by NdibeRaymond — COMMENTED** [review]
Another thing I wasn't able to comment on directly because the changes are way to much for github to render is the fact that you have so many dirty diffs in the package-lock.json file. This is probably related to the stuff about the issue with your npm version and also maybe related to windows vs linux differences. We should work on fixing this before it can be merged.

**Comment by Morai-C** [comment]
@tuxology @NdibeRaymond  i just added another commit to address the issues and it works fine now without any errors.. in the new commit i changed the react-scripts and react_quill  back to the default version and also added the utils folder which my npm modified when i ran the npm install command.. everything works fine with npm now and no errors. please review thank you

File: zubhub_frontend/zubhub/package.json [hunk 4]
```diff
@@ -26,14 +26,15 @@
...
-    "react-scripts": "4.0.3",
+    "react-scripts": "^4.0.3",
     "react-slick": "^0.27.13",
     "react-toastify": "^6.1.0",
     "redux": "^4.0.5",
     "redux-persist": "^6.0.0",
     "redux-thunk": "^2.3.0",
     "slick-carousel": "^1.8.1",
     "source-map-explorer": "^2.5.2",
+    "util": "^0.12.4",
```
Comment (kamthamc) [inline zubhub_frontend/zubhub/package.json]: Please don’t add libraries. Please revert the changes to package.json & package-lock.json

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 2 above)
Comment (kamthamc) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: @tuxology  Any screen size less than [900px](https://mui.com/customization/breakpoints/) will have 0 marginLeft and marginRight.

@Morai-C You don't need to specify units if the value is zero. It can be simply `marginLeft: 0`.

If you are setting margin left and right values you could also use `margin-inline`
https://developer.mozilla.org/en-US/docs/Web/CSS/margin-inline

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js [hunk 5]
```diff
@@ -118,8 +124,9 @@ const styles = theme => ({
     marginTop: '0.5em',
     marginBottom: '0.5em',
     display: 'flex',
-    justifyContent: 'flex-start',
+    justifyContent: 'space-between',
```
Comment (kamthamc) [inline zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: Please check this in different resolutions. It might look weird if there is too much spacing

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 2 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: @kamthamc thanks for this.. i'm guessing the margin-inline would work the same way marginHorizontal on react native works?

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 5 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: yeah did that before adjusting it and after adjusting..

File: zubhub_frontend/zubhub/package.json (hunk 4 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/package.json]: when i first ran npm install it did some adjustments and the util was part of that.. so it started causing errors because the util was in use but the dependency was no longer there... so i reverted back to the default packages but it still didn't get the util but the errors reduced to just the util so i added it and all the errors cleared cos it could now see the module. anyone running frontend on npm without that would be getting lots of errors and reload issues if their npm messes with that..

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 2 above)
Comment (kamthamc) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: It should work in the same way. `margin-inline` is a standard css property. And I think `marginHorizontal` is specific to react-native

File: zubhub_frontend/zubhub/package.json (hunk 4 above)
Comment (kamthamc) [reply zubhub_frontend/zubhub/package.json]: I tried to install without docker and I see the package-lock.json was created with old npm and there are some changes in the file. We can create separate issue for updating NodeJS version (in docker) and prerequisites and also update package-lock.json

Comment (kamthamc) [reply zubhub_frontend/zubhub/package.json]: @tuxology

**Comment by Morai-C** [comment]
Description of commit
Closes #367 

Changes
Addresses problem of longer email addresses causing overflow on the profile page, also adjusted reponsiveness of grid items clustering on specific screen resolutions, adjusted comment placeholder text size on smaller screens.
Screenshot (152)
Screenshot (148)
Screenshot (149)
Screenshot (150)
Screenshot (151)

@tuxology  i added a new commit addressing @AndrewLester findings and also made minor adjustments on the responsiveness please review.

File: zubhub_frontend/zubhub/package.json (hunk 4 above)
Comment (tuxology) [reply zubhub_frontend/zubhub/package.json]: Yes, this is a good idea. I'll let @NdibeRaymond Create this one and describe what to exactly do

**Comment by tuxology** [comment]
This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?

**Comment by Morai-C** [comment]
> This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?

please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology

**Comment by NdibeRaymond** [comment]
> > This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?
> 
> please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology

delete the package-lock in your branch, run `git checkout master -- package-lock.json` to pull the original package-lock from master then commit and push

**Comment by Morai-C** [comment]
> > > This looks good and fixes seem to work @Morai-C I am still unsure what do we do with the `package-lock.json` issue for this PR. I suggest you checkout the `package-lock.json` file from master and commit so we can proceed to merge this. Is that the correct workaround @NdibeRaymond?
> > 
> > 
> > please with the package-lock issue i could delete my package-lock and get back another from the master branch and commit or i could run npm install --package-lock-only to resolve conflicts if there are any... which do you suggest i go for? thank you @NdibeRaymond @tuxology
> 
> delete the package-lock in your branch, run `git checkout master -- package-lock.json` to pull the original package-lock from master then commit and push

@tuxology @NdibeRaymond  okay thanks. did that and it says everything is up to date. could you please review and call my attention if there's any issue again. Thank you

**Comment by NdibeRaymond** [comment]
@Morai-C I updated package-lock.json and package.json myself and tested your PR but there are still certain improvements that needs to be made here. The project tag doesn't appear to work well under certain conditions, the images below:
creator_tag_issue
creator_tag_issue_2

It'd be better if the tag could wrap under the username, instead of changing the layout of the project card. Also, you can see the `justify center` stuff doesn't look that great.

**Comment by NdibeRaymond** [comment]
are you still interested in working on this issue @Morai-C ?

**Comment by Morai-C** [comment]
> are you still interested in working on this issue @Morai-C ?

sure thanks @NdibeRaymond i'll get on this right away

**Comment by Morai-C** [comment]
> @Morai-C I updated package-lock.json and package.json myself and tested your PR but there are still certain improvements that needs to be made here. The project tag doesn't appear to work well under certain conditions, the images below: creator_tag_issue creator_tag_issue_2
> 
> It'd be better if the tag could wrap under the username, instead of changing the layout of the project card. Also, you can see the `justify center` stuff doesn't look that great.

okay i'll do exactly that @NdibeRaymond

**Comment by Morai-C** [comment]
@NdibeRaymond @tuxology  i just committed again with the latest changes, made the tag wrap due to lengthy usernames, removed the justify content and restored the layout back to its default.. i took off the margins and added grid spacing instead due to the fact that with 3 columns in the grid and mui breakpoints having different values from the original breakpoints when testing the responsiveness the cards touch each and caused minor overlaps.. please review thank you.

Screenshot (181)
Screenshot (176)
Screenshot (177)
Screenshot (178)
Screenshot (180)

**Comment by tuxology** [comment]
This looks ok to me now. I think its time to merge this @NdibeRaymond ?

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js [hunk 6]
```diff
@@ -118,13 +120,23 @@ const styles = theme => ({
...
+    width: '100%',
   },
   creatorAvatarStyle: {
     marginRight: '0.5em',
     boxShadow: `0 3px 5px 2px rgba(0, 0, 0, .12)`,
   },
+  creatorEmailAndTagBoxStyle: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    [theme.breakpoints.down('376')]: {
```
Comment (kamthamc) [inline zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: Its better to use predefined breakpoints like sm or md 

https://mui.com/material-ui/customization/breakpoints/

**Review by kamthamc — COMMENTED** [review]
I think its better to use predefined breakpoints instead of adding different values at random places. Apart from it LGTM

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 6 above)
Comment (Morai-C) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js:133]: with the pre-defined breakpoints, the values for the sm or md are quite high so in order to get the wrap to work without changes, I created the box and used that style just for that line.

Comment (Morai-C) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: ok @kamthamc @tuxology  i'll do that now and push

**Comment by Morai-C** [comment]
@tuxology please i've made the changes

File: zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js (hunk 6 above)
Comment (kamthamc) [reply zubhub_frontend/zubhub/src/assets/js/styles/components/project/projectStyles.js]: If we want to use custom size we could define a different size while creating the theme and use it here

**Comment by NdibeRaymond** [comment]
@Morai-C thanks for all the effort you put into this PR. Though I still feel like there are changes that still need to be made. There are certain changes made to the project card that I don't fully understand the rationale behind. The project card width was reduced for no obvious reason. There is padding bottom on the username even when the user tag is not wrapped. I feel like I need to properly take time to look at the css here.

**Comment by Morai-C** [comment]
> @Morai-C thanks for all the effort you put into this PR. Though I still feel like there are changes that still need to be made. There are certain changes made to the project card that I don't fully understand the rationale behind. The project card width was reduced for no obvious reason. There is padding bottom on the username even when the user tag is not wrapped. I feel like I need to properly take time to look at the css here.

@NdibeRaymond @tuxology  thanks for giving me the chance to work on this I am happy to help anytime. I feel the padding-bottom or margin-bottom is a result of the p component by default the p tag has a padding-bottom or margin-bottom.  I adjusted the card because given the allocation of predefined breakpoints when adjusting the browser size to check responsiveness on certain widths with the cards taking 3 columns they either overlapped or were joined without any gap while trying to maintain the width of the parent container. so to maintain the spacing I adjusted the card width and also added a gap. should in case you need me to clarify any other thing let me know.

**Comment by tuxology** [comment]
@NdibeRaymond The breakpoints and card width change does make sense to me but I would let you decide if we should be ok with this since it might break UI in multiple places. It is not breaking it on my large and small screens at least. If you feel it is ok, we can request @Morai-C to rebase and resolve conflicts so we can merge?

**Comment by tuxology** [comment]
Closing this since card design has substantially changed
